src/client/serialize/daos_serialize.c

```diff
@@ -397,11 +377,13 @@ daos_cont_serialize_md(char *filename, daos_prop_t *props, int num_attrs,
 
 	D_PRINT("Writing metadata file: %s\n", filename);
 
+	if (filename == NULL)
+		D_GOTO(out, rc = -DER_INVAL);
+
```

Comment:
This would need to be before the line above that prints the filename, otherwise there is still risk of dereferencing a NULL pointer

Reply:
@daltonbohning oops, thanks.